sdk/core/azure-core/test/ut/operation_test.hpp

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <azure/core/context.hpp>
+#include <azure/core/nullable.hpp>
+#include <azure/core/operation.hpp>
+#include <azure/core/operation_status.hpp>
+#include <azure/core/response.hpp>
+#include <memory>
+#include <string>
+#include <thread>
+
+namespace Azure { namespace Core { namespace Test {
+
+  class StringClient;
+
+  class StringOperation : public Operation<std::string> {
+
+  private:
+    StringClient* m_client;
+    std::string m_operationToken;
+    std::string m_value;
+
+  private:
+    int m_count = 0;
+
+  private:
+    std::unique_ptr<Http::RawResponse> PollInternal(Context& context) override
+    {
+      // Artificial delay to require 2 polls
+      if (++m_count == 2)
+      {
+        m_status = OperationStatus::Succeeded;
+        m_value = "StringOperation-Completed";
+      }
+
+      // The contents of the response are irrelevant for testing purposes
+      // Need only ensure that a RawResponse is returned
+      return std::make_unique<Http::RawResponse>(
+          (uint16_t)1, (uint16_t)0, Http::HttpStatusCode(200), "OK");
+    }
+
+    Response<std::string> PollUntilDoneInternal(std::chrono::milliseconds period, Context& context)
+        override
+    {
+      std::unique_ptr<Http::RawResponse> response;
+      while (!Done())
+      {
+        // Sleep for the period
+        // Actual clients should respect the retry after header if present
+        std::this_thread::sleep_for(period);
+
+        response = Poll(context);
+      }
+
+      return Response<std::string>(m_value, std::move(response));
```

Comment:
Should we reset `m_count` to 0 at this point, so that a second call to `PollInternal` still waits for a second?

Reply:
No, this is a test not an actual operation.  That behavior, like m_count is an implementation detail